### ISSUE TITLE
Search should be case-insensitive

### ISSUE THREAD

**Issue by dlobron** [dlobron/ChugBot#31]
To make it easier to find existing campers, all search functions should be case-insensitive.  PHP and JavaScript both have lots of options for this, e.g., lowercase everything, or add a case-insensitive option to search calls.

**Comment by brianeroth** [comment]
@dlobron I'm looking at this now and actually not seeing the issue. When I search for a camper that's in the DB on /camperHome.php (`Update Existing`), I can use a range of casings for the first and last name, and it pulls up my record fine. Is there another place where this isn't working?

**Comment by dlobron** [comment]
@brianeroth, I found the same thing!  I actually can't find any place where it's case-sensitive.  Let me check with Simon now- I'll CC you on the email.

**Comment by brianeroth** [comment]
@dlobron I took another look at the special punctuation (apostrophes, hyphens, etc.) based on our email chain, and it does appear that the search won't find the camper if they signed up with a hyphen or apostrophe in their name, but we run the search without it (and vice versa). It's relatively easy to ignore those special characters during the search, but we should think about whether that's something we want the search to do.

**Comment by dlobron** [comment]
@brianeroth , thanks!  Just to make sure I understand: are you saying that if the camper signs up as, e.g., O'Reilly, then we won't find them if they search by "OReilly"?  Does it work if the search does include the punctuation, i.e., if we search for "O'Reilly"? If the latter works, then I think we're OK for now, but let's confirm with Simon when we have our conference.

**Comment by brianeroth** [comment]
@dlobron That's correct. If we search with the apostrophe and the camper signed up with an apostrophe, the search works fine.

**Comment by dlobron** [comment]
Thanks, @brianeroth , for confirming.  I think we can leave this ticket as-is for now.  I'll plan to ask Simon how he wants to handle this, but I think this current functionality is reasonable, and as you say, we may not necessarily want to change it
